ts/features/fci/saga/index.ts

```diff
@@ -262,3 +278,41 @@ function* watchFciEndSaga(): SagaIterator {
     CommonActions.navigate(ROUTES.MAIN)
   );
 }
+
+/**
+ * Handle the FCI polling saga
+ * This saga is used to poll the QTSP filled_document to check if the
+ * document is ready to be downloaded by the user. The polling is done
+ * every POLLING_FREQ_TIMEOUT seconds and the polling is stopped when
+ * the document is ready or when the polling is retried
+ * MAX_POLLING_RETRY times.
+ */
+function* watchFciPollSaga(): SagaIterator {
```

Comment:
Instead of calling the saga repeatedly and storing the retrycounter in the state you can use this approach https://www.supertorio.dev/polling-with-redux-sagas